android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt

```diff
@@ -135,6 +135,37 @@ open class ScreenContainer<T : ScreenFragment>(context: Context?) : ViewGroup(co
         performUpdatesNow()
     }
 
+    private fun findFragmentManagerForReactRootView(rootView: ReactRootView): FragmentManager {
+        var context = rootView.context
+
+        // ReactRootView is expected to be initialized with the main React Activity as a context but
+        // in case of Expo the activity is wrapped in ContextWrapper and we need to unwrap it
+        while (context !is FragmentActivity && context is ContextWrapper) {
+            context = context.baseContext
+        }
+
+        check(context is FragmentActivity) {
+            "In order to use RNScreens components your app's activity need to extend ReactActivity"
+        }
+
+        // In case React Native is loaded on a Fragment (not directly in activity) we need to find
+        // fragment manager whose fragment's view is ReactRootView. As of now, we detect such case by
+        // checking whether any fragments are attached to activity which hosts ReactRootView.
+        // See: https://github.com/software-mansion/react-native-screens/issues/1506 on why the cases
+        // must be treated separately.
+        return if (context.supportFragmentManager.fragments.isEmpty()) {
+            // We are in standard React Native application w/o custom native navigation based on fragments.
+           context.supportFragmentManager
+        } else {
+            // We are in some custom setup & we want to use the closest fragment manager in hierarchy.
+            // `findFragment` method throws IllegalStateException when it fails to resolve appropriate
+            // fragment. It might happen when e.g. React Native is loaded directly in Activity
+            // but some custom fragments are still used. However such use case seems highly unlikely
+            // so, as for now, we let application crash.
+            FragmentManager.findFragment<Fragment>(rootView).childFragmentManager
```

Comment:
Potentially we could catch the exception here and fallback to `context.supportFragmentManager`:

```suggestion
            // We are in some custom setup & we want to use the closest fragment manager in hierarchy.
            // `findFragment` method throws IllegalStateException when it fails to resolve appropriate
            // fragment. It might happen when e.g. React Native is loaded directly in Activity
            // but some custom fragments are still used. However such use case seems highly unlikely
            // so, as for now, we let application crash.
            try { 
                FragmentManager.findFragment<Fragment>(rootView).childFragmentManager
            } catch (IllegalStateException e) {
                context.supportFragmentManager
            }
```

Reply:
Change added with [714ddea](https://github.com/software-mansion/react-native-screens/pull/1553/commits/714ddea0dfafe421a5aacac42e23b165fe770afc)